src/routes/CreateSafePage/components/SafeCreationProcess.tsx

```diff
@@ -78,102 +87,115 @@ const parseError = (err: Error): Error => {
   return actualMessage ? new Error(actualMessage) : err
 }
 
-function SafeCreationProcess(): ReactElement {
-  const [safeCreationTxHash, setSafeCreationTxHash] = useState<string | undefined>()
-  const [creationTxPromise, setCreationTxPromise] = useState<Promise<TransactionReceipt>>()
+const getSavedSafeCreation = (): CreateSafeFormValues | void => {
+  return loadFromStorage<CreateSafeFormValues>(SAFE_PENDING_CREATION_STORAGE_KEY)
+}
 
-  const { trackEvent } = useAnalytics()
-  const dispatch = useDispatch()
-  const userAddressAccount = useSelector(userAccountSelector)
-  const chainId = useSelector(currentChainId)
+const loadSavedDataOrLeave = (): CreateSafeFormValues | void => {
+  return getSavedSafeCreation() || goToWelcomePage()
+}
 
-  const [showModal, setShowModal] = useState(false)
-  const [modalData, setModalData] = useState<ModalDataType>({ safeAddress: '' })
-  const [showCouldNotLoadModal, setShowCouldNotLoadModal] = useState(false)
-  const [newSafeAddress, setNewSafeAddress] = useState<string>('')
+const createNewSafe = (userAddress: string, onHash: (hash: string) => void): Promise<TransactionReceipt> => {
+  if (!userAddress) {
+    return Promise.reject(new Error('No user address'))
+  }
 
-  const createNewSafe = useCallback(() => {
-    const safeCreationFormValues = loadFromStorage<CreateSafeFormValues>(SAFE_PENDING_CREATION_STORAGE_KEY)
+  const safeCreationFormValues = loadSavedDataOrLeave()
```

Comment:
I understand the logic, but this is a bit confusing to follow. I would check `safeCreationFormValues` here instead of coupling it with loading.

```suggestion
  const safeCreationFormValues = getSavedSafeCreation()
  if (!safeCreationFormValues) {
    goToWelcomePage()
  }
```

Reply:
Not sure I follow.

Reply:
I would assume `safeCreationFormValues` has data on face value.

I'm not sure about the combination of loading/navigating away in `loadSavedDataOrLeave`.

Reply:
I still don't get what you want me to do.
I merely encapsulated duplicate pieces of code into a function.
This code was repeated three times:

```
    const safeCreationFormValues = loadFromStorage<CreateSafeFormValues>(SAFE_PENDING_CREATION_STORAGE_KEY)
    if (!safeCreationFormValues) {
      goToWelcomePage()
      return
    }
```

https://github.com/gnosis/safe-react/blob/dev/src/routes/CreateSafePage/components/SafeCreationProcess.tsx#L96-L101
https://github.com/gnosis/safe-react/blob/dev/src/routes/CreateSafePage/components/SafeCreationProcess.tsx#L154-L158
https://github.com/gnosis/safe-react/blob/dev/src/routes/CreateSafePage/components/SafeCreationProcess.tsx#L169-L174

Now it's in a function called `loadSavedDataOrLeave`.